### PR TITLE
Prevent duplication of text avatar into display name

### PR DESCRIPTION
**Pull request by BetaRays**
Fixes https://github.com/conversejs/converse.js/issues/3529.

- [x] Add a changelog entry for your change in `CHANGES.md`
- ~~When adding a configuration variable, please make sure to
      document it in `docs/source/configuration.rst`~~
- [ ] Please add a test for your change. Tests can be run in the commandline
      with `make check` or you can run them in the browser by running `make serve`
      and then opening `http://localhost:8000/tests.html`.

### CODE REVIEW

**Comment by BetaRays** [comment]
I tried making a test for it, but I’m not sure how to wait until the name has been updated, if it’s supposed to not change.

**Comment by BetaRays** [comment]
Also, I’m not sure if adding an entry to the changelog is necessary here if the bug didn’t appear in the previous version (or any released version).

File: src/plugins/roomslist/view.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ export class RoomsList extends CustomElement {
     async openRoom (ev) {
         ev.preventDefault();
         const target = u.ancestor(/** @type {HTMLElement} */(ev.target), '.open-room');
-        const name = target.textContent;
+        const name = target.querySelector(".room-name")?.textContent;
```
Comment (jcbrand) [inline src/plugins/roomslist/view.js:58]: Thank you for the fix.

I think a slightly better solution is to put the name as a `data-` attribute on the `a` element, similar to `data-room-jid`.

Then this becomes: `const name = target.getAttribute('data-room-name');`

**Comment by jcbrand** [comment]
> I tried making a test for it, but I’m not sure how to wait until the name has been updated, if it’s supposed to not change.

Maybe just look for an existing test in the roomslist plugin and add a check that the name stays the same when clicking on the item in the list to open a room.

**Comment by BetaRays** [comment]
> > I tried making a test for it, but I’m not sure how to wait until the name has been updated, if it’s supposed to not change.
> 
> Maybe just look for an existing test in the roomslist plugin and add a check that the name stays the same when clicking on the item in the list to open a room.

My issue is that I need to put some kind of delay between clicking and checking for the new name (otherwise the test passes when it shouldn’t because the name hasn’t been updated yet): this is usually done with `u.waitUntil` but I’m not sure what condition to use (waiting for the name to change would fail if it stays the same as it’s supposed to).

**Comment by jcbrand** [comment]
> My issue is that I need to put some kind of delay between clicking and checking for the new name (otherwise the test passes when it shouldn’t because the name hasn’t been updated yet): this is usually done with `u.waitUntil` but I’m not sure what condition to use (waiting for the name to change would fail if it stays the same as it’s supposed to).

You could set a different name before clicking and then checking that it is set back to the right name, but I don't think a test is that important in this case, so I'll merge.

**Pull request merged**